pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -44,11 +46,12 @@ const finalizer = "foreground-deletion"
 // Add creates a new PerformanceProfile Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+	r := newReconciler(mgr)
+	return add(mgr, r, r.ppRequestsFromMCP)
```

Comment:
@slintes  The problem is that` r reconcile.Reconciler` which add gets:
https://github.com/openshift-kni/performance-addon-operators/blob/20b946decfb3e4d8f14b28e326efe38bd3b14c7c/pkg/controller/performanceprofile/performanceprofile_controller.go#L64
Is not a `ReconcilePerformanceProfile` and I'm not eager to change the type here because I assume it should stay an interface here. 
WDYF ?

Reply:
Hm, I see. But changing type isn't a problem IMHO, `add()` isn't public and only called by `Add()`